File: conformance/bundles/14-stateless-steps/squares.steps.py
from var import define_state

# No state factory: these steps are pure, so define_state() is called bare
# and handlers get an empty dict as state.
stimulus, sensor = define_state()


@stimulus("I warm up my mental math")
def _(state):
    pass


@sensor("The square of {int} is {int}.")
def _(state, n, expected):
    return [n, n * n]

File: conformance/bundles/14-stateless-steps/squares.steps.rb
require "oselvar/var"

# No state factory: these steps are pure, so steps is called bare and handlers
# get an empty hash as state.
param, stimulus, sensor = steps

stimulus.("I warm up my mental math") { |_state| }

sensor.("The square of {int} is {int}.") { |_state, n, _expected| [n, n * n] }
